src/geometry/manifold/algebra/monoid.lean

```diff
@@ -112,6 +114,39 @@ lemma smooth_on.mul {f : M → G} {g : M → G} {s : set M}
   smooth_on I' I (f * g) s :=
 ((smooth_mul I).comp_smooth_on (hf.prod_mk hg) : _)
 
+variables (I) (g h : G)
+
+/-- Left multiplication by `g`. It is meant to mimic the usual notation in Lie groups. -/
+def smooth_left_mul : C^∞⟮I, G; I, G⟯ := ⟨(left_mul g), smooth_mul_left⟩
+
+/-- Left multiplication by `g`. It is meant to mimic the usual notation in Lie groups. -/
+def smooth_right_mul : C^∞⟮I, G; I, G⟯ := ⟨(right_mul g), smooth_mul_right⟩
+
+/- Left multiplication. The abbreviation is `MIL`. -/
+localized "notation `𝑳` := smooth_left_mul" in lie_group
+
+/- Left multiplication. The abbreviation is `MIL`. -/
+localized "notation `𝑹` := smooth_right_mul" in lie_group
+
+open_locale lie_group
+
+@[simp] lemma L_apply : (𝑳 I g) h = g * h := rfl
+@[simp] lemma R_apply : (𝑹 I g) h = h * g := rfl
+
+@[simp] lemma L_mul {G : Type*} [semigroup G] [topological_space G] [charted_space H G]
+  [has_smooth_mul I G] (g h : G) : 𝑳 I (g * h) = (𝑳 I g).comp (𝑳 I h) :=
+by { ext, simp only [times_cont_mdiff_map.comp_apply, L_apply, mul_assoc] }
+
+@[simp] lemma R_mul {G : Type*} [semigroup G] [topological_space G] [charted_space H G]
+  [has_smooth_mul I G] (g h : G) : 𝑹 I (g * h) = (𝑹 I h).comp (𝑹 I g) :=
+by { ext, simp only [times_cont_mdiff_map.comp_apply, R_apply, mul_assoc] }
+
+lemma L_apply_one {G : Type*} [monoid G] [topological_space G] [charted_space H G]
+  [has_smooth_mul I G] (g : G) : (𝑳 I g) 1 = g := by rw [L_apply, mul_one]
+
+lemma R_apply_one {G : Type*} [monoid G] [topological_space G] [charted_space H G]
```

Comment:
These could also be simp lemmas.

Reply:
Both lemmas should be proven by simp, so they should not be simp lemmas. However, in my Lie algebras simp did not solve the lemmas that made use of them, so I added them. I will take them out of this PR and add them back if the problem still persists. It was a problem one year ago, maybe something changed for the simp in the while...